solutions/USACO jan20-berry-picking.mdx

```diff
@@ -0,0 +1,139 @@
+---
+id: usaco-990
+source: USACO Silver 2020 January
+title: Berry Picking
+author: Qi Wang
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_berries_silver_jan20.html)
+
+**Time Complexity:** $\mathcal{O}(max(B_i) \cdot N\log N)$
+
+
+<LanguageSection>
+<JavaSection>
+
+```java
+//Created by Qi Wang
+import java.util.*;
+import java.io.*;
+
+public class berries {
+    static int N, K;
+    static int[] B;
+    public static void main(String[] args) throws IOException {
+        InputReader in = new InputReader("berries.in");
+        PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("berries.out")));
+        N = in.nextInt(); K = in.nextInt(); B = new int[N];
+        int max = -1;
+        for (int i = 0; i < N; i++) {
+            B[i] = in.nextInt();
+            max = Math.max(max, B[i]);
+        }
+
+        int res = -1;
+        for (int i = 1; i <= max; i++) {
+            int count = 0;
+            int[] leftOver = new int[N];
+            for (int j = 0; j < N; j++) {
+                count += B[j]/i;
+                leftOver[j] = B[j]%i;
+            }
+
+            if(count >= K){
+                //If count can account for both Bessie and Elsie, then we can just allocate K/2 * i to Bessie
+                res = Math.max(res, K/2 * i);
+            }
+            else if(count >= K/2){
+                //If count can only account of Elsie and some of Bessie, then we can take from the left overs.
```

Comment:
Typo "account of Elsie" and "left overs"

Probably should be
`//If count can only account for Elsie and some of Bessie, then we can take from the leftovers.`